pretender.js

```diff
@@ -178,24 +179,66 @@ Pretender.prototype = {
 
     if (handler) {
       handler.handler.numberOfCalls++;
+      var async = handler.handler.async;
       this.handledRequests.push(request);
 
       try {
         var statusHeadersAndBody = handler.handler(request),
             status = statusHeadersAndBody[0],
             headers = this.prepareHeaders(statusHeadersAndBody[1]),
-            body = this.prepareBody(statusHeadersAndBody[2]);
-        request.respond(status, headers, body);
+            body = this.prepareBody(statusHeadersAndBody[2]),
+            pretender = this;
 
-        this.handledRequest(verb, path, request);
+        this.handleResponse(request, async, function() {
+          request.respond(status, headers, body);
+          pretender.handledRequest(verb, path, request);
+        });
       } catch (error) {
         this.erroredRequest(verb, path, request, error);
+        pretender.resolve(request);
       }
     } else {
       this.unhandledRequests.push(request);
       this.unhandledRequest(verb, path, request);
     }
   },
+  requestRefs: [],
+  handleResponse: function handleResponse(request, strategy, fn) {
+    strategy = typeof strategy === 'function' ? strategy() : strategy;
+
+    if (strategy === false) {
+      fn();
+    } else {
+      var pretender = this;
+      pretender.requestRefs.push({
+        request: request,
+        fn: fn
+      });
+
+      if (strategy !== true) {
+        setTimeout(function() {
+          pretender.resolve(request);
+        }, typeof strategy === 'number' ? strategy : 0);
+      }
+    }
+  },
+  resolve: function resolve(request) {
```

Comment:
`fn` is an ambiguous variable name :stuck_out_tongue: 

It's the callback that handles the response. I will rename it to something better.